### PR TITLE
Bug 1841572 – removed extra spacing from the bookmark menu item

### PR DESCRIPTION
**Pull request by mavduevskiy**
There were a bunch of bugs duplicating this [one](https://bugzilla.mozilla.org/show_bug.cgi?id=1823645), because we didn't have enough space for the text in bookmark checkbox button. 

Initially I was just planning to remove the container end spacing because checkbox also have padding of its own. But then I realized that the start spacing for the button is 32pd minimum, which is, given the constraints of what we are trying to fit it, looks like an overkill. Reducing it to 16 helped to fit add button for the most of the languages. I have seen some edit buttons still trenched, but we probably shouldn't make padding less than 16dp.

#### Welsh
 Old | New
 --- | --- 
 |  
#### German
 Old | New
 --- | --- 
 |  
#### Greek
 Old | New
 --- | --- 
 |  
#### English
 Old | New
 --- | --- 
 |  

### Pull Request checklist

- [ ] **Quality**: This PR builds and passes detekt/ktlint checks (A pre-push hook is recommended)
- [ ] **Tests**: This PR includes thorough tests or an explanation of why it does not
- [ ] **Changelog**: This PR includes [a changelog entry](https://github.com/mozilla-mobile/firefox-android/blob/main/docs/changelog.md) or does not need one
- [ ] **Accessibility**: The code in this PR follows [accessibility best practices](https://github.com/mozilla-mobile/firefox-android/blob/main/docs/shared/android/accessibility_guide.md) or does not include any user facing features

### After merge
- **Breaking Changes**: If this is a breaking Android Components change, please push a draft PR on [Reference Browser](https://github.com/mozilla-mobile/reference-browser) to address the breaking issues.

### To download an APK when reviewing a PR (after all CI tasks finished running):
1. Click on `Checks` at the top of the PR page.
2. Click on the `firefoxci-taskcluster` group on the left to expand all tasks.
3. Click on the `build-apk-{fenix,focus,klar}-debug` task you're interested in.
4. Click on `View task in Taskcluster` in the new `DETAILS` section.
5. The APK links should be on the right side of the screen, named for each CPU architecture.

### GitHub Automation
https://bugzilla.mozilla.org/show_bug.cgi?id=1841572

### CODE REVIEW

**Comment by mavduevskiy** [comment]
@Verdi 
It looks like a minor thing, but just want to be sure that there is no hard NO from the UX team :)

File: android-components/components/browser/menu/src/main/res/layout/mozac_browser_menu_item_image_text_checkbox_button.xml
Comment (mavduevskiy) [inline android-components/components/browser/menu/src/main/res/layout/mozac_browser_menu_item_image_text_checkbox_button.xml:60]: splitting 16dp padding in half, so buttons have equal tapping area in the worst case with very long strings in both buttons.

File: android-components/components/browser/menu/src/main/res/layout/mozac_browser_menu_item_image_text_checkbox_button.xml
Comment (gabrielluong) [inline android-components/components/browser/menu/src/main/res/layout/mozac_browser_menu_item_image_text_checkbox_button.xml:32]: Because this is an Android Components, we don't want to hardcode these padding values like this. The styles are meant to be configurable by the consumer. https://searchfox.org/mozilla-mobile/source/firefox-android/android-components/components/browser/menu/README.md#18

File: android-components/components/browser/menu/src/main/res/layout/mozac_browser_menu_item_image_text_checkbox_button.xml
Comment (gabrielluong) [inline android-components/components/browser/menu/src/main/res/layout/mozac_browser_menu_item_image_text_checkbox_button.xml:13]: Could you explain what this change is doing? Why not paddingHorizontal?

Comment (mavduevskiy) [reply android-components/components/browser/menu/src/main/res/layout/mozac_browser_menu_item_image_text_checkbox_button.xml:13]: Now, reading your other comment, I realize that probably I should be doing that by overriding the dimens as well. 

But anyway, intent is to remove end container spacing, and delegate the end spacing for the checkbox button.
That's the structure of spacing that we have now: 
`> >text_view< >checkbox< <`. 

And that's what I am going for:
`> >text_view< >checkbox<`.

File: android-components/components/browser/menu/src/main/res/layout/mozac_browser_menu_item_image_text_checkbox_button.xml
Comment (mavduevskiy) [reply android-components/components/browser/menu/src/main/res/layout/mozac_browser_menu_item_image_text_checkbox_button.xml:32]: It looks like doing it through the @styles would require some additional work, to decouple checkbox button from the regular menu button. Those use the same styling for container and for the text view. At the same time, checkbox style doesn't have spacing in its style...

Since the change is needed on the fenix side, how about we extend BrowserMenuImageTextCheckboxButton and provide a new xml in`getLayoutResource`?

**Comment by Verdi** [comment]
> @Verdi It looks like a minor thing, but just want to be sure that there is no hard NO from the UX team :)

Yep - this is fine

**Comment by mavduevskiy** [comment]
@gabrielluong 
Implemented by adding new styles and overriding those on the fenix side. If the approach aligns with best AC practices, I will proceed with updating the changlog.

**Comment by gabrielluong** [comment]
@mavduevskiy I'm still on PTO - you might want to find someone to review this in the meantime. Maybe @rocketsroger or @mcarare?

File: fenix/app/src/main/res/values/dimens.xml
Comment (rocketsroger) [inline fenix/app/src/main/res/values/dimens.xml:14]: Why do we need to add these?

Comment (rocketsroger) [inline fenix/app/src/main/res/values/dimens.xml:14]: Why do we need to add these?

**Review by rocketsroger — COMMENTED** [review]
Works well.  I do have a question though.

File: fenix/app/src/main/res/values/dimens.xml
Comment (mavduevskiy) [reply fenix/app/src/main/res/values/dimens.xml:14]: It's kinda the same [question](https://github.com/mozilla-mobile/firefox-android/pull/2697#discussion_r1252732024).

In short, the container has end spacing, and the checkbox button has end spacing, so the text doesn't get enough space. I extended styling on A-C level, while keeping the existing design, so we could override things we want to look different on the fenix level. 

Basically, I am removing end spacing for the container for the element that has a text and a checkbox button.

Comment (rocketsroger) [reply fenix/app/src/main/res/values/dimens.xml:14]: No I mean why is it `tools:ignore="UnusedResources"`?

File: android-components/components/browser/menu/src/main/res/values/style.xml
Comment (gabrielluong) [inline android-components/components/browser/menu/src/main/res/values/style.xml:79]:
```suggestion
        <item name="android:layout_width">0dp</item>
```

File: android-components/components/browser/menu/src/main/res/values/style.xml
Comment (gabrielluong) [inline android-components/components/browser/menu/src/main/res/values/style.xml:72]:
```suggestion
        <item name="android:layout_width">wrap_content</item>
```
Some of your <item>s are not aligned.

File: fenix/app/src/main/res/values/dimens.xml
Comment (gabrielluong) [inline fenix/app/src/main/res/values/dimens.xml:17]:
```suggestion
    <dimen name="mozac_browser_menu_item_image_checkbox_padding_start" tools:ignore="UnusedResources">8dp</dimen>
    <dimen name="mozac_browser_menu_item_image_checkbox_padding_end" tools:ignore="UnusedResources">16dp</dimen>
```

Could we reorder these so that the image checkbox paddings are next to eachother and the same for the checkbox text label paddings?

File: android-components/components/browser/menu/src/main/res/values/style.xml
Comment (gabrielluong) [inline android-components/components/browser/menu/src/main/res/values/style.xml:71]: Should we not inherit from `Mozac.Browser.Menu.Item.ImageText.Label` here?  This is what you're replacing above and you could remove the reductant `android:layout_width` and `android:layout_height`. Note, `Mozac.Browser.Menu.Item.ImageText.Label` already inherits from from `Mozac.Browser.Menu.Item.Text`.

**Review by gabrielluong — APPROVED** [review]

**Comment by gabrielluong** [comment]
@mavduevskiy Before we land this, I was also wondering if the values used in this PR came from any figma or did you come up with the values? If it came from a figma, it would be nice to link it somewhere here or ideally in the bugzilla so someone looking back at this can see where the designs are coming from. Otherwise, if you came up with the values, then I believe your original PR description should be sufficient. The point is to just make it explicit where these values are coming from for the people looking back at this. Everything is good since we already got the approval from UX.

**Comment by gabrielluong** [comment]
One last thing, I did test on the a RTL language, but perhaps it is also worthwhile for you to test to make sure the spacing is what you expected if you haven't done so.

**Comment by mavduevskiy** [comment]
Hey @gabrielluong 

Thanks for pointing the inheritance thing, it was really useful. I often avoid it when coding business logic, and didn't realize that styling is one of the areas we should be striving for it. Your comments were spot on👍

I checked a few RTL languages, they are surprisingly concise.

For the sake of bookkeeping: there is no FIGMA design for this change, it was just me looking for an optimal implementation of the x8 matrix in a limited space of this component.

**Pull request merged**
